<a href="https://colab.research.google.com/github/ameya932/world_happiness_report/blob/main/World_Happiness_Report.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

In [1]:
from google.colab import drive
drive.mount('/content/drive')

Mounted at /content/drive


In [2]:
import numpy as np
import pandas as pd

In [4]:
path = "/content/drive/MyDrive/DataForFigure2.1WHR2021C2.xls"
df = pd.read_excel(path)

**DATA PREPARATION AND CLEANING**

In [5]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 149 entries, 0 to 148
Data columns (total 20 columns):
 #   Column                                      Non-Null Count  Dtype  
---  ------                                      --------------  -----  
 0   Country name                                149 non-null    object 
 1   Regional indicator                          149 non-null    object 
 2   Ladder score                                149 non-null    float64
 3   Standard error of ladder score              149 non-null    float64
 4   upperwhisker                                149 non-null    float64
 5   lowerwhisker                                149 non-null    float64
 6   Logged GDP per capita                       149 non-null    float64
 7   Social support                              149 non-null    float64
 8   Healthy life expectancy                     149 non-null    float64
 9   Freedom to make life choices                149 non-null    float64
 10  Generosity    

In [6]:
df.describe()

Ladder score  Standard error of ladder score  upperwhisker  \
count    149.000000                      149.000000    149.000000   
mean       5.532832                        0.058764      5.648008   
std        1.073923                        0.022011      1.054320   
min        2.522900                        0.025583      2.596468   
25%        4.851800                        0.043369      4.991074   
50%        5.533500                        0.054181      5.624688   
75%        6.255400                        0.069863      6.344270   
max        7.842100                        0.172664      7.904126   

       lowerwhisker  Logged GDP per capita  Social support  \
count    149.000000             149.000000      149.000000   
mean       5.417655               9.432209        0.814726   
std        1.094875               1.158585        0.114892   
min        2.449332               6.635322        0.462596   
25%        4.705666               8.540956        0.749777   
50%        5.413106               9.568728        0.831914   
75%        6.127992              10.421198        0.904702   
max        7.780075              11.646564        0.982938   

       Healthy life expectancy  Freedom to make life choices  Generosity  \
count               149.000000                    149.000000  149.000000   
mean                 64.992784                      0.791572   -0.015149   
std                   6.762071                      0.113315    0.150653   
min                  48.478001                      0.381749   -0.288153   
25%                  59.801552                      0.718051   -0.126469   
50%                  66.603096                      0.804414   -0.036025   
75%                  69.599998                      0.877263    0.078830   
max                  76.952858                      0.970131    0.541553   

       Perceptions of corruption  Ladder score in Dystopia  \
count                 149.000000                149.000000   
mean                    0.727477                  2.430343   
std                     0.179266                  0.000000   
min                     0.081959                  2.430343   
25%                     0.666607                  2.430343   
50%                     0.780623                  2.430343   
75%                     0.844619                  2.430343   
max                     0.939343                  2.430343   

       Explained by: Log GDP per capita  Explained by: Social support  \
count                        149.000000                    149.000000   
mean                           0.977153                      0.793315   
std                            0.404777                      0.258840   
min                            0.000000                      0.000000   
25%                            0.665774                      0.646990   
50%                            1.024849                      0.832038   
75%                            1.322677                      0.996021   
max                            1.750786                      1.172279   

       Explained by: Healthy life expectancy  \
count                             149.000000   
mean                                0.520132   
std                                 0.212971   
min                                 0.000000   
25%                                 0.356634   
50%                                 0.570849   
75%                                 0.665236   
max                                 0.896814   

       Explained by: Freedom to make life choices  Explained by: Generosity  \
count                                  149.000000                149.000000   
mean                                     0.498687                  0.178091   
std                                      0.137885                  0.098277   
min                                      0.000000                  0.000000   
25%                                      0.409224                  0.105473   
50%              



* The Ranking consists of 149 countries and there are no null values.




In [7]:
df.sample()

Country name  Regional indicator  Ladder score  \
95        Niger  Sub-Saharan Africa        5.0741   

    Standard error of ladder score  upperwhisker  lowerwhisker  \
95                        0.101609      5.273253      4.874947   

    Logged GDP per capita  Social support  Healthy life expectancy  \
95               7.097615        0.641138                53.780006   

    Freedom to make life choices  Generosity  Perceptions of corruption  \
95                      0.806124     0.01845                   0.692541   

    Ladder score in Dystopia  Explained by: Log GDP per capita  \
95                  2.430343                          0.161512   

    Explained by: Social support  Explained by: Healthy life expectancy  \
95                      0.402237                               0.166986   

    Explained by: Freedom to make life choices  Explained by: Generosity  \
95                                    0.516394                  0.200009   

    Explained by: Perceptions of corruption  Dystopia + residual  
95                                 0.157392             3.469616